modules.d/90multipath/module-setup.sh

```diff
@@ -41,11 +41,9 @@ depends() {
 
 # called by dracut
 cmdline() {
-    for m in scsi_dh_alua scsi_dh_emc scsi_dh_rdac dm_multipath; do
-        if grep -m 1 -q "$m" /proc/modules; then
-            printf 'rd.driver.pre=%s ' "$m"
-        fi
-    done
+    if grep -m 1 -q dm_multipath /proc/modules; then
+        printf 'rd.driver.pre=%s ' dm_multipath
+    fi
 }
```

Comment:
I wonder is this is required at all; I'd have thought that multipathd should be able to load 'dm-multipath' on its own. But until then I'm fine with this patch.

Reply:
You are right - we've got an `ExecStartPre=-/sbin/modprobe dm-multipath` in the unit file, which should be sufficient. 

We can't start multipathd without loading `dm-multipath` beforehand, because multipathd makes libdm calls to figure out the version of the multipath target, which will fail if the module isn't loaded. While implenting that functionality in `multipathd` itself would be simple, it's still easier to keep the `ExecStartPre` line.

@johannbg, if it's ok for you I'll change that.

Reply:
Modprobe should never be called from a systemd unit so that `ExecStartPre=` should be removed not kept or used.

Reply:
I've never heard this before. Can you provide any backing /reasoning for this statement? What's wrong with calling modprobe in `ExecStartPre`?

Reply:
Interesting I thought that was common knowledge by now since it's been like a decade ago since I went throught this whole dialog when handling sysv to systemd migration ( along with the whole remove modules etc )  anyways modules should be auto-loaded using MODALIAS and or similar constructs in the kernel modules + udev or if people really needed to load a module statically, they should do so via /etc/modules-load.d/ but this should never be done in type unit files in systemd. 

There should never be any need to load a module statically these days unless for debugging/testing purposes, if there is any need beside that something is broken.